R/powerEmp.R

```diff
@@ -0,0 +1,114 @@
+#' Log-Rank Test for a Single Trial
+#'
+#' This function evaluates the significance of either PFS or OS endpoints in a trial,
+#' based on a pre-specified critical value.
+#'
+#' @param data (`data.frame`)\cr data frame containing entry and exit times of an
+#' illness-death model. See [getSimulatedData()] for details.
+#' @param typeEvent (`string`)\cr endpoint to be evaluated, possible values are `PFS` and `OS`.
+#' @param critical (positive `number`)\cr critical value of the log-rank test.
+#'
+#' @return Logical value indicating log-rank test significance.
+#' @export
+#'
+#' @examples
+#' library(survival)
+#' transition1 <- exponential_transition(h01 = 0.06, h02 = 0.3, h12 = 0.3)
+#' transition2 <- exponential_transition(h01 = 0.1, h02 = 0.4, h12 = 0.3)
+#' simTrial <- getClinicalTrials(
+#'   nRep = 1, nPat = c(800, 800), seed = 1234, datType = "1rowPatient",
+#'   transitionByArm = list(transition1, transition2), dropout = list(rate = 0.5, time = 12),
+#'   accrual = list(param = "intensity", value = 7)
+#' )[[1]]
+#' logRankTest(data = simTrial, typeEvent = "OS", critical = 3.4)
+logRankTest <- function(data, typeEvent, critical) {
+  # must be `1rowTransition`, i.e. have 10 (censored) or 11 columns
+  assert_data_frame(data, min.cols = 10, max.cols = 11)
+  assert_choice(typeEvent, c("OS", "PFS"))
+  assert_positive_number(critical)
+
+  if (typeEvent == "OS") {
+    time <- data$OStime
+    event <- data$OSevent
+  } else if (typeEvent == "PFS") {
+    time <- data$PFStime
+    event <- data$PFSevent
+  }
+
+  logRank <- survdiff(Surv(time, event) ~ trt, data)
```

Comment:
```suggestion
  logRank <- survival::survdiff(Surv(time, event) ~ trt, data)
```
since it is not in base R